pymc3/gp/cov.py

```diff
@@ -88,7 +92,8 @@ def __array_wrap__(self, result):
 class Combination(Covariance):
     def __init__(self, factor_list):
         input_dim = np.max([factor.input_dim for factor in
-                            filter(lambda x: isinstance(x, Covariance), factor_list)])
+                            filter(lambda x: isinstance(x, Covariance),
```

Comment:
This approach seems like it would let you collapse the nested iterations into a single comprehension as well.